packages/edit-post/src/index.js

```diff
@@ -97,6 +98,7 @@ export function initializeEditor(
 		welcomeGuideTemplate: true,
 	} );
 
+	dispatch( blocksStore ).__experimentalReapplyBlockTypeFilters();
```

Comment:
So if my reading is correct, this function applies the filters already loaded to the blocks that are already loaded? If we call it before `registerCoreBlocks` it won't have any effect right? What am I missing?

Reply:
Yes, it won’t have impact on core blocks. It’s targeting only 3rd party blocks that were registered earlier. In particular, it addresses the issue where 3rd party block filters get registered after 3rd party blocks, example:
- filter A
- block B
- block C
- filter D
- filter E
- block F
- filter G

The new action reset all blocks to the initial state and applies all filters that were missed because of the accidental order of 3rd party plugins.

In the perfect world, we would wait with applying all filters until the block editor gets initialized. We can try it as a next step but it’s hard to tell what implications it would have on the projects that work with the current flow.

Reply:
Makes sense. I think in my ideal world, adding a "block" filter should be treated just like an "action" to the "blocks" store and directly update the blocks objects in the store. The problem I guess is that the "filters" is a separate package.

Reply:
Filters can be also removed which isn't impossible but definitely harder to translate to the actions system. Although, it's essentially what we have for block variations and block styles. I tend to agree that it could play nice if we could build this API from scratch 😄